src/realm/array.cpp

```diff
@@ -358,7 +314,12 @@ void Array::preset(int64_t min, int64_t max, size_t num_items)
 }
 
 
-void Array::destroy_children(size_t offset) noexcept
+void Array::destroy_children() noexcept
+{
+    destroy_some_children(0);
+}
+
+void Array::destroy_some_children(size_t offset) noexcept
```

Comment:
Fair enough. I will come up with a better name. I think the name `destroy_children` implies that all children will be destroyed. I want to make it more explicit when we mean all and when we mean children after a given offset.